crates/rbuilder/src/utils/provider_factory_reopen.rs

```diff
@@ -61,13 +71,23 @@ impl<DB: Database + Clone> ProviderFactoryReopener<DB> {
     /// This will check if historical block hashes for the given block is correct and if not it will reopen
     /// provider fatory.
     /// This should be used when consistency is required: e.g. building blocks.
-    pub fn check_consistency_and_reopen_if_needed(
-        &self,
-        current_block_number: u64,
-    ) -> eyre::Result<ProviderFactory<DB>> {
+    ///
+    /// If the current block number is already known at the time of calling this method, you may pass it to
+    /// avoid an additional DB lookup for the latest block number.
+    pub fn check_consistency_and_reopen_if_needed(&self) -> eyre::Result<ProviderFactory<DB>> {
+        let best_block_number = self
+            .provider_factory_unchecked()
+            .best_block_number()
```

Comment:
What does best block number means and how its different from the latest block number?